registry/registry.go

```diff
@@ -2,80 +2,37 @@
 package registry
 
 import (
-	"context"
-	"encoding/base64"
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"net/http"
-	"net/http/cookiejar"
+	"github.com/go-kit/kit/log"
 	"sort"
-	"strconv"
-	"strings"
 	"time"
-
-	"github.com/docker/distribution/manifest/schema1"
-	"github.com/go-kit/kit/log"
-	dockerregistry "github.com/heroku/docker-registry-client/registry"
-	"github.com/jonboulle/clockwork"
-	"golang.org/x/net/publicsuffix"
-
-	"github.com/weaveworks/flux"
-	fluxmetrics "github.com/weaveworks/flux/metrics"
 )
 
 const (
-	dockerHubHost    = "index.docker.io"
-	dockerHubLibrary = "library"
-
 	requestTimeout = 10 * time.Second
 )
 
-type creds struct {
-	username, password string
-}
-
-// Credentials to a (Docker) registry.
-type Credentials struct {
-	m map[string]creds
-}
-
-// Client is a handle to a bunch of registries.
-type Client interface {
-	GetRepository(repository string) ([]flux.ImageDescription, error)
+// New registry interface
+type Registry interface {
+	GetRepository(repository Repository) ([]Image, error)
+	GetImage(repository Repository, tag string) (Image, error)
 }
 
-// client is a handle to a registry.
-type client struct {
-	Credentials    Credentials
-	Logger         log.Logger
-	Metrics        Metrics
-	MemcacheClient MemcacheClient
-	CacheExpiry    time.Duration
+// registry is a handle to a registry.
```

Comment:
eh.. what? Doesn't seem to be... I'm not really sure what this comment is trying to tell me.